pass options through to the factory

ccutrer opened the pull request:
specifically so that HashRing configuration can be passed through

tubbo commented:
we seem to be getting these `RuntimeError: Node ID collision` errors in test, can you take a look when you get a chance @ccutrer?

ccutrer commented:
I updated it. Somehow extract_options! was working different in older/newer ruby versions. The original commit for using dup (c7a668b8299fb5b4551058bb9af4b60dc85ffc24) is also fixed by my change, so the dup is no longer necessary.

ccutrer commented:
... and updated again to resolve a merge conflict
